### PR TITLE
Migrations: Fix some scaffolding bugs

### LINKED ISSUE #3388: Migrations: Wrong migration generates when DbContext implementation is generic
Subject.
If I try dnx . ef migrations add with generic context implementation, model snapshot generate wrong attribute
[DbContext(typeof(SiteContext`17))]

### PR DESCRIPTION
**Pull request by bricelam**
Highlights:
* Stops escaping contextual C# keywords out of context
* Handle generic types (fixes #3388)

### CODE REVIEW

**Comment by ajcvickers** [comment]
:shipit:

File: src/EntityFramework.Commands/Migrations/Design/CSharpHelper.cs [hunk 1]
```diff
@@ -40,101 +40,60 @@ public class CSharpHelper
             "__reftype",
             "__refvalue",
             "abstract",
-            "add",
-            "alias",
             "as",
-            "ascending",
-            "assembly",
-            "async",
-            "await",
```
Comment (divega) [inline src/EntityFramework.Commands/Migrations/Design/CSharpHelper.cs]: Careful with await. It is contextual but is in context in many places where the others aren't.

Comment (bricelam) [inline src/EntityFramework.Commands/Migrations/Design/CSharpHelper.cs]: The only things we currently generate identifiers for (using this code) are class names in their definition and inside `typeof`.

Comment (divega) [inline src/EntityFramework.Commands/Migrations/Design/CSharpHelper.cs]: Sounds great then.

**Pull request merged**